src/iotjs_string.h

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2015-2016 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IOTJS_STRING_H
+#define IOTJS_STRING_H
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+typedef struct {
+  char* _data;
+  unsigned _size;
+  unsigned _cap;
+} IOTJS_VALIDATED_STRUCT(iotjs_string_t);
```

Comment:
@LaszloLango I think in this case every members are fixed-size data, not header. Anyway I moved the data to the end.
@zherczeg I agree with you. I removed underscores from member variable names. (And after that, I changed simulated-`this` into `_this` to avoid confusion with c++ `this`)